src/test_driver/openiotsdk/integration-tests/common/pyedmgr_device.py

```diff
@@ -0,0 +1,97 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import logging
+import re
+from pyedmgr import TestDevice
+import asyncio
+
+from .device import Device
+
+log = logging.getLogger(__name__)
+
+
+class PyedmgrDevice(Device):
+
+    def __init__(self, device: TestDevice, name=None):
+        super(PyedmgrDevice, self).__init__(name)
+        self.device = device
+        self.run = False
+        self.verbose = True
+
+    async def start(self):
+        """
+        Start the IO task
+        """
+        self.run = True
+        self.input_task = asyncio.create_task(self._input_task())
+        self.output_task = asyncio.create_task(self._output_task())
+
+    async def stop(self):
+        """
+        Stop the IO tasks
+        """
+        self.run = False
+        self.input_task.cancel()
+        self.input_task = None
+        self.output_task.cancel()
+        self.output_task = None
+
+    async def _input_task(self):
+        while self.run:
+            try:
+                line = await asyncio.wait_for(self.device.channel.readline_async(), timeout=0.2)
+                if isinstance(line, bytes) or isinstance(line, bytearray):
+                    line = line.decode("utf-8", errors="replace").strip()
+                line = re.sub(r'\033\[((?:\d|;)*)([a-zA-Z])', '', line)
+                if line:
+                    if self.verbose:
+                        log.info('<--|{}| {}'.format(self.name, line.strip()))
+                    await self.iq.put(line)
+                else:
+                    pass
+            except asyncio.exceptions.TimeoutError:
+                continue
+            except Exception as e:
+                log.debug(f"input thread ({self.name}) loop exception: {e}")
+                log.debug(type(e))
+                log.debug(e.args)
+                log.debug(e)
```

Comment:
Why do we need so detailed log here and below?

Reply:
I got exceptions happening while developing the solution. just printing `e` was not enough. It is unlikelly to happen again as everything is working as intended but I kept the logs in case as it can be quite tricky to find what is wrong in asynchronous code. 

What do you want me to do ?

Reply:
You could use ```traceback.format_exc()``` to get the full exception message as a  string.

Reply:
I was just curious about the reason for this login. But maybe it's worth using Chris's solution to simplify this.

Reply:
this has been fixed. thanks to both of you.